packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -303,26 +286,124 @@ export class OdspDocumentService implements IDocumentService {
         });
     }
 
+    private clearJoinSessionTimer() {
+        if (this.joinSessionRefreshTimer !== undefined) {
+            clearTimeout(this.joinSessionRefreshTimer);
+            this.joinSessionRefreshTimer = undefined;
+        }
+    }
+
+    private async scheduleJoinSessionRefresh(delta: number) {
+        await new Promise<void>((resolve) => {
+            this.joinSessionRefreshTimer = setTimeout(async () => {
+                await getWithRetryForTokenRefresh(async (options) => {
```

Comment:
same feedback as before - errors here will become unhandled rejection promises. Please catch and report them to telemetry.

Reply:
I am catching and reporting to the telemetry in where the scheduleJoinSessionRefresh is called after your previous suggestion. Please check.